Userland/Libraries/LibCrypto/Hash/SHA2.cpp

```diff
@@ -65,18 +67,29 @@ inline void SHA256::transform(u8 const* data)
     m_state[7] += h;
 }
 
-void SHA256::update(u8 const* message, size_t length)
+template<size_t BlockSize, typename Callback>
+void update_buffer(u8* buffer, u8 const* input, size_t length, size_t& data_length, Callback callback)
 {
-    for (size_t i = 0; i < length; ++i) {
-        if (m_data_length == BlockSize) {
-            transform(m_data_buffer);
-            m_bit_length += 512;
-            m_data_length = 0;
+    while (length > 0) {
+        size_t copy_bytes = AK::min(length, BlockSize - data_length);
+        AK::TypedTransfer<u8>::copy(buffer + data_length, input, copy_bytes);
+        length -= copy_bytes;
+        data_length += copy_bytes;
+        if (data_length == BlockSize) {
+            callback();
+            data_length = 0;
         }
-        m_data_buffer[m_data_length++] = message[i];
     }
 }
 
+void SHA256::update(u8 const* message, size_t length)
+{
+    update_buffer<BlockSize>(m_data_buffer, message, length, m_data_length, [&]() {
+        transform(m_data_buffer);
+        m_bit_length += BlockSize * 8;
```

Comment:
(it seems itʼs changed this way in all the use places—is this a rule?)

Reply:
Yes, that's why I changed it to be consistent in the first place. Note that methods and member variables are not inherited  and the base class in templatized, so that's probably why the code is duplicate.

Reply:
oh! i just thought itʼd be nice to abstract that some how (so you could just pass `transform` method only as a parameter instead)

feel free to resolve this comment!